src/modules/form/hooks/rhf/useDynamicFieldWatchValues.tsx

```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react';
+import { FieldValues, useFormContext, useWatch } from 'react-hook-form';
+
+export const useDynamicFieldWatchValues = (watchFields: string[]) => {
+  const { control } = useFormContext();
+  const valueArray = useWatch({
+    control,
+    name: watchFields,
+  });
```

Comment:
```suggestion
export const useDynamicFieldWatchValues = (watchFields?: string[]) => {
  const { control } = useFormContext();
  const valueArray = useWatch({
    control,
    name: watchFields || [],
  });
```
I noticed a performance issue, and this fixes it. Passing `undefined` here causes fields that have no dependencies to watch ALL other fields.

Underlying type issue is that `LinkIdMap` doesn't allow for undefined value, but the `autofillInvertedDependencyMap` doesn't contain all link ids (or have non-undefined values for all link ids).

So either:
* Update map(s) to include all link ids mapped to `[]` when no dependencies (this would be an adjustment to buildAutofillDependencyMap I think, and maybe the inver logic?), or
* Add secondary type like `PartialLinkIdMap = Record<string, string[] | undefined>;` to use for autofill map which we know contains only partial set of keys

Second option seems simpler to me.

Reply:
Thanks for finding and fixing that; I agree that improving the types is probably simplest, I'll do that